lib/SourceMapDevToolPlugin.js

```diff
@@ -2,170 +2,164 @@
-					});
-				} else {
-					sourceMap.sourcesContent = undefined;
-				}
-				sourceMap.sourceRoot = options.sourceRoot || "";
-				sourceMap.file = file;
-				asset.__SourceMapDevToolFile = file;
-				asset.__SourceMapDevToolData = {};
-				var currentSourceMappingURLComment = sourceMappingURLComment;
-				if(currentSourceMappingURLComment !== false && /\.css($|\?)/i.test(file)) {
-					currentSourceMappingURLComment = currentSourceMappingURLComment.replace(/^\n\/\/(.*)$/, "\n/*$1*/");
-				}
-				if(sourceMapFilename) {
-					var filename = file,
-						query = "";
-					var idx = filename.indexOf("?");
-					if(idx >= 0) {
-						query = filename.substr(idx);
-						filename = filename.substr(0, idx);
+					sourceMap.sourceRoot = options.sourceRoot || "";
+					sourceMap.file = file;
+					asset.__SourceMapDevToolFile = file;
+					asset.__SourceMapDevToolData = {};
+					let currentSourceMappingURLComment = sourceMappingURLComment;
+					if(currentSourceMappingURLComment !== false && /\.css($|\?)/i.test(file)) {
+						currentSourceMappingURLComment = currentSourceMappingURLComment.replace(/^\n\/\/(.*)$/, "\n/*$1*/");
 					}
-					var sourceMapFile = this.getPath(sourceMapFilename, {
-						chunk: chunk,
-						filename: filename,
-						query: query,
-						basename: basename(filename)
-					});
-					var sourceMapUrl = path.relative(path.dirname(file), sourceMapFile).replace(/\\/g, "/");
-					if(currentSourceMappingURLComment !== false) {
-						asset.__SourceMapDevToolData[file] = this.assets[file] = new ConcatSource(new RawSource(source), currentSourceMappingURLComment.replace(/\[url\]/g, sourceMapUrl));
+					if(sourceMapFilename) {
+						let filename = file;
+						let query = "";
+						const idx = filename.indexOf("?");
+						if(idx >= 0) {
+							query = filename.substr(idx);
+							filename = filename.substr(0, idx);
+						}
+						const sourceMapFile = compilation.getPath(sourceMapFilename, {
+							chunk,
+							filename,
+							query,
+							basename: basename(filename)
+						});
+						const sourceMapUrl = path.relative(path.dirname(file), sourceMapFile).replace(/\\/g, "/");
+						if(currentSourceMappingURLComment !== false) {
+							asset.__SourceMapDevToolData[file] = compilation.assets[file] = new ConcatSource(new RawSource(source), currentSourceMappingURLComment.replace(/\[url\]/g, sourceMapUrl));
+						}
+						asset.__SourceMapDevToolData[sourceMapFile] = compilation.assets[sourceMapFile] = new RawSource(JSON.stringify(sourceMap));
+						chunk.files.push(sourceMapFile);
+					} else {
+						asset.__SourceMapDevToolData[file] = compilation.assets[file] = new ConcatSource(new RawSource(source), currentSourceMappingURLComment
+							.replace(/\[map\]/g, () => JSON.stringify(sourceMap))
+							.replace(/\[url\]/g, () => `data:application/json;charset=utf-8;base64,${Buffer.from(JSON.stringify(sourceMap)).toString("base64")}`)
```

Comment:
`Buffer.from` is not supported in older node.js versions. Use `new Buffer` instead.